File: backend/app/schemas/evidence.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class EvidenceItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    competitor_name: str | None
    source_url: str
    source_title: str | None
    source_type: str | None
    chunk_index: int
    chunk_text: str
    reliability_score: Decimal | None
    created_at: datetime


class EvidenceListResponse(BaseModel):
    items: list[EvidenceItem]

File: frontend/src/types/evidence.ts
export interface EvidenceItem {
  id: number
  task_id: number
  competitor_name?: string | null
  source_url: string
  source_title?: string | null
  source_type?: string | null
  chunk_index: number
  chunk_text: string
  reliability_score?: string | number | null
  created_at: string
}
